src/components/GooglePlacesAutocomplete.jsx

```diff
@@ -0,0 +1,127 @@
+import React, { useCallback, useEffect, useMemo, useRef, useState } from "react";
+import PropTypes from "prop-types";
+import { startCase, debounce } from "lodash";
+import { Input } from "./Forms/Input";
+import { Badge } from "./Badge";
+
+export const GooglePlacesAutocomplete = ({ initialValue, onSelect, urlConfig }) => {
+    const [inputValue, setInputValue] = useState(initialValue || "");
+    const [suggestions, setSuggestions] = useState([]);
+    const [isLoading, setIsLoading] = useState(false);
+    const [showDropdown, setShowDropdown] = useState(false);
+
+    const dropdownRef = useRef(null);
+    const initialFetchDoneRef = useRef(false);
+
+    const handleSelect = useCallback(
+        (sug) => {
+            setInputValue(sug.description || sug.name || "");
+            setShowDropdown(false);
+            onSelect?.(sug);
+        },
+        [onSelect],
+    );
+
+    const handleInputChange = (e) => {
+        setInputValue(e.target.value);
+        setShowDropdown(true);
+        fetchSuggestions(e.target.value);
+    };
+
+    const fetchSuggestions = useMemo(
+        () =>
+            debounce(async (query, selectFirst = false) => {
```

Comment:
Both `useMemo` and `debounce` will not work, its inefficient. Use the `useDebouncedCallback` from `use-debounce` It's in the seller app